### PR TITLE
[complex] `torch.sigmoid`: CUDA support and complex autograd support

### LINKED ISSUE #48552: [complex] torch.sigmoid: sigmoid does not support automatic differentiation for outputs with complex dtype
```python
import torch
x = torch.tensor(1, dtype=torch.cfloat)
x.requires_grad = True
y = torch.sigmoid(x)
y.backward()
```

**Output**:
```
Traceback (most recent call last):
  File "test/test_exp.py", line 16, in <module>
    y = torch.sigmoid(x)
RuntimeError: sigmoid does not support automatic differentiation for outputs with complex dtype.
```

cc @ezyang @anjali411 @dylanbespalko @mruberry @Lezcano

### PR DESCRIPTION
**Pull request by kshitij12345**
Fixes https://github.com/pytorch/pytorch/issues/48552

**Changes**

* Complex support for `torch.sigmoid` CUDA (CPU support already exists)
* Complex autograd support for `torch.sigmoid` (CUDA and CPU)

### CODE REVIEW

File: tools/autograd/derivatives.yaml [hunk 1]
```diff
@@ -1707,8 +1707,8 @@
   grad_output: upsample_nearest3d(grad, output_size, scale_factors)
 
 - name: sigmoid_backward(Tensor grad_output, Tensor output) -> Tensor
-  grad_output: sigmoid_backward(grad, output)
-  output: grad * grad_output * (-2 * output + 1)
+  grad_output: sigmoid_backward(grad, output.conj())
```
Comment (anjali411) [inline tools/autograd/derivatives.yaml:1710]: why do you pass `output.conj()` here?

**Comment by anjali411** [comment]
@kshitij12345 can you please rebase?

**Review by anjali411 — APPROVED** [review]

**Comment by kshitij12345** [comment]
@anjali411 Apologies not sure why it worked locally. (Will ping you once all the CI passes). Thanks!

**Comment by kshitij12345** [comment]
@anjali411 this is ready now. Failures look irrelevant.

**Comment by kshitij12345** [comment]
@anjali411 gentle ping :)

**Pull request closed without merging**